libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceRestAPITests.m

```diff
@@ -314,7 +314,7 @@ - (void)testCreateQuerySearchDelete {
 // - query updated object (requestForSearch) and make sure we don't get anything
 - (void)testCreateUpdateQuerySearchDelete {
     // create object
-    NSString *lastName = [NSString stringWithFormat:@"Doe-%@", [NSDate date]];
```

Comment:
This is our bug, because the encoding of the query is the responsibility of the Mobile SDK.  It's actually a regression brought in from [this PR](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1754).

@bhariharan , in `CSFPrivateDefines` (and elsewhere in that PR), we've replaced the usage of `CFURLCreateStringByAddingPercentEscapes()` with `stringByAddingPercentEncodingWithAllowedCharacters:`, which is fine, but `[NSCharacterSet URLQueryAllowedCharacterSet]` isn't a complete set of characters to escape, unfortunately.  We should be doing a superset of that set and what we had before.

From [this SO answer](http://stackoverflow.com/a/29810106), I'd say that set looks like `#%<>[\]^``{|}&:/=+`.